### ISSUE TITLE
A crash report

### ISSUE THREAD

**Issue by runminlook** [yandex/ClickHouse#5515]
(you don't have to strictly follow this form)

**Describe the bug**
A crash report.

**How to reproduce**
Restart ClickHouse then crash again.

* Which ClickHouse server version to use
19.6.2.1

* Which interface to use, if matters
* Non-default settings, if any
* `CREATE TABLE` statements for all tables involved
```sql
ATTACH TABLE login_ch1
(
    `d` Date, 
    `appId` Int32, 
    `pluginId` Int32, 
    `ms` Int64, 
    `msDelay` Int32, 
    `msLogin` Int64, 
    `userId` Int32, 
    `gid` Int32, 
    `userIp` IPv4, 
    `userCity` String, 
    `clientType` Int32, 
    `isGuildMember` Int8, 
    `senderId` Int32, 
    `senderIp` IPv4
)
ENGINE = MergeTree()
PARTITION BY d
PRIMARY KEY (appId, pluginId, toStartOfMinute(toDateTime(ms / 1000)))
ORDER BY (appId, pluginId, toStartOfMinute(toDateTime(ms / 1000)), ms)
TTL d + toIntervalDay(10)
SETTINGS index_granularity = 8192


ALTER TABLE login_ch1 ADD COLUMN IF NOT EXISTS msDelay Int32 AFTER ms;
ALTER TABLE login_ch1 ADD COLUMN IF NOT EXISTS msLogin Int64 AFTER msDelay;
ALTER TABLE login_ch1 ADD COLUMN IF NOT EXISTS userCity String AFTER userIp;

ALTER TABLE login_ch1 MODIFY TTL d + toIntervalDay(10); // origin 3 days
```

**Error message and/or stacktrace**

```
2019.06.03 00:00:06.593767 [ 39 ] {} <Debug> default.login_ch1 (MergerMutator): Selected MergeAlgorithm: Horizontal
2019.06.03 00:00:06.593793 [ 39 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 20190602_48798_48798_0, total 12068 rows starting from the beginning of the part, columns: d, appId, pluginId, ms, msDelay, msLogin, userId, gid, userIp, userCity, clientType, isGuildMember, senderId, senderIp
2019.06.03 00:00:06.594098 [ 39 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 3 marks from part 20190602_48799_48799_0, total 16694 rows starting from the beginning of the part, columns: d, appId, pluginId, ms, msDelay, msLogin, userId, gid, userIp, userCity, clientType, isGuildMember, senderId, senderIp
2019.06.03 00:00:06.594373 [ 39 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 20190602_48800_48800_0, total 9525 rows starting from the beginning of the part, columns: d, appId, pluginId, ms, msDelay, msLogin, userId, gid, userIp, userCity, clientType, isGuildMember, senderId, senderIp
2019.06.03 00:00:06.594649 [ 39 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 2 marks from part 20190602_48801_48801_0, total 8933 rows starting from the beginning of the part, columns: d, appId, pluginId, ms, msDelay, msLogin, userId, gid, userIp, userCity, clientType, isGuildMember, senderId, senderIp
2019.06.03 00:00:06.594903 [ 39 ] {} <Trace> MergeTreeSequentialBlockInputStream: Reading 1 marks from part 20190602_48802_48802_0, total 7174 rows starting from the beginning of the part, columns: d, appId, pluginId, ms, msDelay, msLogin, userId, gid, userIp, userCity, clientType, isGuildMember, senderId, senderIp
2019.06.03 00:00:06.623542 [ 39 ] {} <Debug> default.login_ch1 (MergerMutator): Merge sorted 54394 rows, containing 14 columns (14 merged, 0 gathered) in 0.03 sec., 1812014.11 rows/sec., 162.46 MB/sec.
2019.06.03 00:00:06.625995 [ 39 ] {} <Trace> default.login_ch1 (Data): Renaming temporary part tmp_merge_20190602_48798_48802_1 to 20190602_48798_48802_1.
2019.06.03 00:00:06.626161 [ 39 ] {} <Trace> default.login_ch1 (MergerMutator): Merged 5 parts: from 20190602_48798_48798_0 to 20190602_48802_48802_0
2019.06.03 00:00:06.832401 [ 52 ] {} <Error> BaseDaemon: ########################################
2019.06.03 00:00:06.984072 [ 52 ] {} <Error> BaseDaemon: (version 19.6.2.1) (from thread 29) Received signal Segmentation fault (11).
2019.06.03 00:00:06.984140 [ 52 ] {} <Error> BaseDaemon: Address: NULL pointer.
2019.06.03 00:00:06.984148 [ 52 ] {} <Error> BaseDaemon: Access: read.
2019.06.03 00:00:06.984154 [ 52 ] {} <Error> BaseDaemon: Address not mapped to object.
2019.06.03 00:00:07.347215 [ 52 ] {} <Error> BaseDaemon: 0. clickhouse-server(DB::Block::bytes() const+0x97) [0x5dde547]
2019.06.03 00:00:07.347263 [ 52 ] {} <Error> BaseDaemon: 1. clickhouse-server(DB::BlockStreamProfileInfo::update(DB::Block&)+0x26) [0x5e2e536]
2019.06.03 00:00:07.347273 [ 52 ] {} <Error> BaseDaemon: 2. clickhouse-server(DB::IBlockInputStream::read()+0x253) [0x5e339c3]
2019.06.03 00:00:07.347293 [ 52 ] {} <Error> BaseDaemon: 3. clickhouse-server(DB::MergeTreeDataMergerMutator::mergePartsToTemporaryPart(DB::FutureMergedMutatedPart const&, DB::MergeListEntry&, long, DB::DiskSpaceMonitor::Reservation*, bool)+0x1a1f) [0x61fd05f]
2019.06.03 00:00:07.347305 [ 52 ] {} <Error> BaseDaemon: 4. clickhouse-server(DB::StorageMergeTree::merge(bool, std::string const&, bool, bool, std::string*)+0x35b) [0x614ecab]
2019.06.03 00:00:07.347322 [ 52 ] {} <Error> BaseDaemon: 5. clickhouse-server(DB::StorageMergeTree::backgroundTask()+0x168) [0x61506a8]
2019.06.03 00:00:07.347331 [ 52 ] {} <Error> BaseDaemon: 6. clickhouse-server(DB::BackgroundProcessingPool::threadFunction()+0x3e8) [0x61b1538]
2019.06.03 00:00:07.347338 [ 52 ] {} <Error> BaseDaemon: 7. clickhouse-server() [0x61b1f34]
2019.06.03 00:00:07.347355 [ 52 ] {} <Error> BaseDaemon: 8. clickhouse-server(ThreadPoolImpl<std::thread>::worker(std::_List_iterator<std::thread>)+0x187) [0x68d7947]
2019.06.03 00:00:07.347361 [ 52 ] {} <Error> BaseDaemon: 9. clickhouse-server() [0x745371f]
2019.06.03 00:00:07.347369 [ 52 ] {} <Error> BaseDaemon: 10. /lib64/libpthread.so.0(+0x7e25) [0x7f4aa231ee25]
2019.06.03 00:00:10.780868 [ 51 ] {} <Trace> TCPHandlerFactory: TCP Request. Address: 10.111.71.64:56132
2019.06.03 00:00:10.781030 [ 51 ] {} <Debug> TCPHandler: Connected ClickHouse python-driver version 18.10.0, revision: 54406, database: default, user: default.
2019.06.03 00:00:10.781545 [ 51 ] {4e99bfbe-8c4f-4563-a0fa-707acec122e9} <Debug> executeQuery: (from 10.111.71.64:56132) INSERT INTO logout_ch1 VALUES
2019.06.03 00:00:11.059177 [ 51 ] {4e99bfbe-8c4f-4563-a0fa-707acec122e9} <Trace> default.logout_ch1 (Data): Renaming temporary part tmp_insert_20190602_48806_48806_0 to 20190602_48806_48806_0.
2019.06.03 00:00:11.059323 [ 51 ] {4e99bfbe-8c4f-4563-a0fa-707acec122e9} <Debug> MemoryTracker: Peak memory usage (for query): 32.16 MiB.
```

**Comment by jackpgao** [comment]
same problem.

**Comment by CurtizJ** [comment]
Fixed in #5819.
